.github/workflows/CD.yml

```diff
@@ -318,11 +319,12 @@ jobs:
           key: "${{ env.PACKAGE_CACHE_KEY }}"
           path: build/packages
 
-      - name: Checkout the sha
+      #Since the tools from workflow are always up-to-date and to keep it up with the github workflow, we need to check out the main branch
+      #And use the tools from there instead of the commited sha branch to avoid commit additional code for the commited sha's branch
+      - name: Checkout main branch to get workflow tools
         uses: actions/checkout@v2
         with:
-          ref: ${{ github.event.inputs.sha }}
-          path: adot
+          ref: "main"
```

Comment:
This is not required. If we trigger the workflow dispatch, it'll use whatever branch is selected, which should be `main`, but can be another branch.

Reply:
Reasonable to me after thinking. Fixed it.